packages/babel-helpers/src/helpers.js

```diff
@@ -238,30 +238,35 @@ helpers.asyncGeneratorDelegate = () => template.program.ast`
 `;
 
 helpers.asyncToGenerator = () => template.program.ast`
+  function asyncGeneratorStep(gen, resolve, reject, _next, _throw, key, arg) {
+    try {
+      var info = gen[key](arg);
+      var value = info.value;
+    } catch (error) {
+      reject(error);
+      return;
+    }
+
+    if (info.done) {
+      resolve(value);
+    } else {
+      Promise.resolve(value).then(_next, _throw);
+    }
+  }
+
   export default function _asyncToGenerator(fn) {
     return function () {
       var self = this, args = arguments;
       return new Promise(function (resolve, reject) {
         var gen = fn.apply(self, args);
-        function step(key, arg) {
-          try {
-            var info = gen[key](arg);
-            var value = info.value;
-          } catch (error) {
-            reject(error);
-            return;
-          }
-
-          if (info.done) {
-            resolve(value);
-          } else {
-            Promise.resolve(value).then(_next, _throw);
-          }
+        function _next(value) {
+          asyncGeneratorStep(gen, resolve, reject, _next, _throw, "next", value);
+        }
+        function _throw(err) {
+          asyncGeneratorStep(gen, resolve, reject, _next, _throw, "throw", err);
         }
-        function _next(value) { step("next", value); }
-        function _throw(err) { step("throw", err); }
 
-        _next();
+        _next(undefined);
```

Comment:
any particular reason for this explicit undefined?

Reply:
Monomorphic calls. Probably negligible.

Reply:
is arg length important for monomorphism? always thought more about shapes in this regard